website/cue/reference/components/sinks/mqtt.cue

```diff
@@ -0,0 +1,142 @@
+package metadata
+
+components: sinks: mqtt: {
+	title: "MQTT"
+
+	classes: {
+		commonly_used: false
+		delivery:      "best_effort"
+		development:   "beta"
+		egress_method: "stream"
+		service_providers: []
+		stateful: false
+	}
+
+	features: {
+		acknowledgements: true
+		healthcheck: enabled: false
+		send: {
+			compression: enabled: false
+			encoding: {
+				enabled: true
+				codec: {
+					enabled: true
+					enum: ["json", "text"]
+				}
+			}
+			request: enabled: false
+			tls: {
+				enabled:                true
+				can_verify_certificate: true
+				can_verify_hostname:    true
+				enabled_default:        false
+			}
+			to: {
+				service: services.mqtt
+				interface: {
+					socket: {
+						direction: "outgoing"
+						protocols: ["tcp"]
+						ssl: "optional"
+					}
+				}
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration: {
+		host: {
+			description: """
+				The MQTT broker to connect to.
+				"""
+			required: true
+			warnings: []
+			type: string: {
+				examples: ["mqtt.example.com"]
+				syntax: "literal"
+			}
+		}
+		port: {
+			description: """
+                                MQTT service port to connect to.
+                                """
+			required: false
+			type: uint: {
+				default: 1883
+			}
+		}
+		user: {
+			description: """
+                                MQTT username
+                                """
+			required: true
+			type: str: {}
+		}
+		password: {
+			description: """
+                                MQTT password
+                                """
+			required: true
+			type: str: {}
+		}
```

Comment:
What are your thoughts on making these not required and default to empty strings?

Reply:
Fixed